src/servers/src/http/event.rs

```diff
@@ -387,6 +359,89 @@ pub async fn pipeline_dryrun(
     Ok(Json(result).into_response())
 }
 
+/// Dryrun pipeline with given data
+/// pipeline_name and pipeline_version to specify pipeline stored in db
+/// pipeline to specify pipeline raw content
+/// data to specify data
+/// data maght be list of string or list of object
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct PipelineDryrunParams {
+    pub pipeline_name: Option<String>,
+    pub pipeline_version: Option<String>,
+    pub pipeline: Option<String>,
+    pub data: Vec<Value>,
+}
+
+#[axum_macros::debug_handler]
+pub async fn pipeline_dryrun(
+    State(log_state): State<LogState>,
+    Query(query_params): Query<LogIngesterQueryParams>,
+    Extension(mut query_ctx): Extension<QueryContext>,
+    TypedHeader(content_type): TypedHeader<ContentType>,
+    payload: String,
+) -> Result<Response> {
+    let handler = log_state.log_handler;
+    fn check_pipeline_name_exists(pipeline_name: Option<String>) -> Result<String> {
+        pipeline_name.context(InvalidParameterSnafu {
+            reason: "pipeline_name is required",
+        })
+    }
+    match serde_json::from_str::<PipelineDryrunParams>(&payload) {
```

Comment:
The matching arms of this match are confusing, why is `Err(_) | Ok(_)` exist on the same path. I suggest refactoring this logic according to each scenario and please add more comments on each case.

Note: the description of this pr seems out of date, please update it too.

Reply:
updated the description.
And a separate function is extracted to determine whether the new logic is used or not.